operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/TemporaryResourceCache.java

```diff
@@ -61,18 +69,34 @@ public synchronized void putResource(T newResource, String previousResourceVersi
         .orElse(managedInformerEventSource.get(resourceId).orElse(null));
 
     if ((previousResourceVersion == null && cachedResource == null)
-        || (cachedResource != null && previousResourceVersion != null
-            && cachedResource.getMetadata().getResourceVersion()
-                .equals(previousResourceVersion))) {
+        || (cachedResource != null
+            && (cachedResource.getMetadata().getResourceVersion().equals(previousResourceVersion))
+            || isLaterResourceVersion(resourceId, newResource, cachedResource))) {
       log.debug(
           "Temporarily moving ahead to target version {} for resource id: {}",
           newResource.getMetadata().getResourceVersion(), resourceId);
       putToCache(newResource, resourceId);
-    } else {
-      if (cache.remove(resourceId) != null) {
-        log.debug("Removed an obsolete resource from cache for id: {}", resourceId);
+    } else if (cache.remove(resourceId) != null) {
+      log.debug("Removed an obsolete resource from cache for id: {}", resourceId);
+    }
+  }
+
+  private boolean isLaterResourceVersion(ResourceID resourceId, T newResource, T cachedResource) {
+    try {
+      if (managedInformerEventSource.getInformerConfiguration() instanceof InformerConfiguration
+          && ((InformerConfiguration) managedInformerEventSource.getInformerConfiguration())
+              .parseResourceVersions()
+          && Long.compare(Long.parseLong(newResource.getMetadata().getResourceVersion()),
+              Long.parseLong(cachedResource.getMetadata().getResourceVersion())) > 1) {
+        return true;
       }
+    } catch (NumberFormatException e) {
+      log.debug(
+          "Could not compare resourceVersions {} and {} for {}",
```

Comment:
Shouldn't be this a warning? even if it could pollute cache? (no strong opinion)

Reply:
It won't pollute the cache, it would exhibit the same behavior as it does now.  It can of course be at whatever level you want.

Reply:
I mean pollute logs :)